packages/rum-react/src/domain/reactRouterV6/registerRouter.ts

```diff
@@ -0,0 +1,16 @@
+import type { createBrowserRouter } from 'react-router-dom'
+import { startReactRouterView } from './startReactRouterView'
+
+type Router = ReturnType<typeof createBrowserRouter>
+
+export function registerRouter(router: Router) {
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏Following my reasoning on the [useRoute instrumentation](https://github.com/DataDog/browser-sdk/pull/2824/files#r1648604979), we could use the same approach using a HOF to be consistent:
```ts
const datadogCreateBrowserRouter = Datadog.createBrowserRouter(createBrowserRouter)
```

Reply:
The `Routes` component and `useRoutes` hook don't provide a way to cleanly subscribe to route changes. Given this constraint, wrapping them is kind of a requirement.

On the contrary, router instances have a way to subscribe to changes, so we don't have this constraint. That being said, for consistency, we could consider exposing a `createBrowserRouter` which wraps the original `createBrowserRouter` and just subscribe to it internally. I think that's interesting! Let me check how this could look like.